prody/ensemble/ensemble.py

```diff
@@ -433,44 +462,37 @@ def addCoordset(self, coords):
             self._confs = coords
         else:
             self._confs = concatenate((self._confs, coords), axis=0)
+        
+        # appending new data
+        all_keys = set(list(self._data.keys()) + list(adddata.keys()))
```

Comment:
You need to only run this if `self._data.keys()` is not `None`. You can maybe have a LOGGER warning if it doesn't.

Otherwise, we get the tests failing as follows:
```
======================================================================
ERROR: Failure: AttributeError ('NoneType' object has no attribute 'keys')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test/lib/python3.8/site-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/home/travis/miniconda/envs/test/lib/python3.8/site-packages/nose/loader.py", line 416, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/home/travis/miniconda/envs/test/lib/python3.8/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/travis/miniconda/envs/test/lib/python3.8/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/travis/miniconda/envs/test/lib/python3.8/imp.py", line 244, in load_module
    return load_package(name, filename)
  File "/home/travis/miniconda/envs/test/lib/python3.8/imp.py", line 216, in load_package
    return _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/travis/miniconda/envs/test/lib/python3.8/site-packages/ProDy-1.11-py3.8-linux-x86_64.egg/prody/tests/ensemble/__init__.py", line 36, in <module>
    PDBENSEMBLE.addCoordset(xyz)
  File "/home/travis/miniconda/envs/test/lib/python3.8/site-packages/ProDy-1.11-py3.8-linux-x86_64.egg/prody/ensemble/pdbensemble.py", line 389, in addCoordset
    all_keys = set(list(self._data.keys()) + list(adddata.keys()))
AttributeError: 'NoneType' object has no attribute 'keys'
```

Reply:
Actually, you need to check that `adddata.keys()` is not `None` too.